connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/ProxyHandler.java

```diff
@@ -0,0 +1,121 @@
+package io.camunda.connector.http.base.client.apache;
+
+import io.camunda.connector.api.error.ConnectorInputException;
+import java.net.*;
+import java.util.Arrays;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hc.client5.http.auth.AuthScope;
+import org.apache.hc.client5.http.auth.CredentialsProvider;
+import org.apache.hc.client5.http.auth.UsernamePasswordCredentials;
+import org.apache.hc.client5.http.impl.DefaultSchemePortResolver;
+import org.apache.hc.client5.http.impl.auth.BasicCredentialsProvider;
+import org.apache.hc.client5.http.impl.routing.DefaultProxyRoutePlanner;
+import org.apache.hc.client5.http.impl.routing.SystemDefaultRoutePlanner;
+import org.apache.hc.client5.http.routing.HttpRoutePlanner;
+import org.apache.hc.core5.http.HttpHost;
+
+public class ProxyHandler {
+
+  record CustomProxy(String host, int port, String username, String password) {}
+
+  public static CredentialsProvider getCredentialsProvider(String uncheckedProtocol) {
+    BasicCredentialsProvider provider = new BasicCredentialsProvider();
+    String protocol = fallbackOnHttpForInvalidProtocol(uncheckedProtocol);
+    try {
+      if (isValidProxyWithAuth(protocol)) {
+        provider.setCredentials(
+            new AuthScope(
+                System.getProperty(protocol + ".proxyHost"),
+                Integer.parseInt(System.getProperty(protocol + ".proxyPort"))),
+            new UsernamePasswordCredentials(
+                System.getProperty(protocol + ".proxyUser"),
+                System.getProperty(protocol + ".proxyPassword").toCharArray()));
+      } else if (systemEnvIsSetForProtocol(protocol)) {
+        CustomProxy proxyData = getProxySettingsFromEnvVar(protocol);
+        provider.setCredentials(
+            new AuthScope(proxyData.host, proxyData.port),
+            new UsernamePasswordCredentials(proxyData.username, proxyData.password.toCharArray()));
+      }
```

Comment:
Would it be possible to have a dedicated `ProxyConfig` data class and a single place (a factory) that is responsible for creating it? 

It would be easier to follow and having a single place where data is read either from System.properties and EnvVars would make it easier to debug.

Reply:
I just saw that we have `CustomProxy` available. Why not use it here as well:
```
        provider.setCredentials(
            new AuthScope(
                System.getProperty(protocol + ".proxyHost"),
                Integer.parseInt(System.getProperty(protocol + ".proxyPort"))),
            new UsernamePasswordCredentials(
                System.getProperty(protocol + ".proxyUser"),
                System.getProperty(protocol + ".proxyPassword").toCharArray()));
```

Reply:
You are right, that would make it easier to read. I will adjust it.